src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/CachedExpressionCompiler.cs

```diff
@@ -61,66 +74,109 @@ private static class Compiler<TModel, TResult>
 
                     // model => model.Member
                     case MemberExpression memberExpression when memberExpression.Expression == expression.Parameters[0]:
-                        return CompileFromMemberAccess(expression, memberExpression);
+                        return CompileFromSimpleMemberAccess(expression, memberExpression);
+
+                    // model => model.Member1.Member2
+                    case MemberExpression memberExpression when IsChainedPropertyAccessor(memberExpression):
+                        return CompileForChainedMemberAccess(expression, memberExpression);
 
                     default:
-                        return CompileSlow(expression);
+                        return null;
+                }
+
+                bool IsChainedPropertyAccessor(MemberExpression memberExpression)
```

Comment:
No need to inline. Suggest making this into another `private static` method

Reply:
Local functions get you 💯 points